### PR TITLE
Refactor tracing subscriber instatiation

### PR DESCRIPTION
**Pull request by prestwich**
- Adds an empty Jaeger configuration struct to the settings
- Refactors and DRY subscriber setup

### CODE REVIEW

File: rust/optics-base/src/settings/trace/jaeger.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+use opentelemetry::{sdk::trace::Tracer, trace::TraceError};
+use opentelemetry_jaeger::PipelineBuilder;
+use tracing::Subscriber;
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::registry::LookupSpan;
+
+/// Config parameters for collection via Jaeger
+#[derive(Debug, Clone, Copy, serde::Deserialize)]
+pub struct JaegerConfig {
+    // TODO
```
Comment (prestwich) [inline rust/optics-base/src/settings/trace/jaeger.rs:10]: @yourbuddyconner check out the docs for the Jaeger tracer here:

https://docs.rs/opentelemetry-jaeger/0.14.0/opentelemetry_jaeger/struct.PipelineBuilder.html#method.with_agent_endpoint

each of those `with_` methods is a configuration option. Please let me know which of them you want/need to be able to configure from env/json. Then I will populate this struct and implement the `into` below

File: rust/optics-base/src/settings/trace/fmt.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+impl Default for Style {
+    fn default() -> Self {
+        Style::Full
+    }
+}
+
+/// Unification of the fmt Subscriber formatting modes
+///
+/// You may be asking yourself, why does this exist. I ask myself the same thing
```
Comment (yourbuddyconner) [inline rust/optics-base/src/settings/trace/fmt.rs:36]: lol

File: rust/optics-base/src/settings/trace/jaeger.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+/// Config parameters for Jaeger collector
+#[derive(Debug, Clone, serde::Deserialize)]
+pub struct JaegerCollector {
+    uri: String,
+    #[serde(flatten)]
+    auth: Option<CollectorAuth>,
+}
+
+/// Config parameters fo
```
Comment (yourbuddyconner) [inline rust/optics-base/src/settings/trace/jaeger.rs:22]: Extra comment

**Review by yourbuddyconner — DISMISSED** [review]
Config LGTM, left a nit for a dangling comment

**Pull request merged**